### PR TITLE
Rspec test

### PR DESCRIPTION
**Pull request by RoodzFernando**
We just wrote some tests on the method in the main file.

### CODE REVIEW

**Comment by shshamim63** [comment]
## Status :recycle:  [!! Required Changes !!] :recycle: 
___
Hi @RoodzFernando  and @luis-novoa  :wave: ,
Well done :clap: , you have tried to cover all the corner to handle the edge cases of this project by writing test. But it is not complete yet. There are some issues that are actually blocking me to declare this project as complete. And I am sure that you will cover them easily :sunglasses: .
___
##### Please follow the instructions  below :point_down: to complete this project.
___
### :spades: [Mandatory Changes] :spades: 
___
1. I can see some lint error on this project. First of all, they are really necessary to take in count.
2. Actually the best and well known way to create a dependent variable is by using let, as far I know. But you have used direct assign operation.
https://github.com/RoodzFernando/Tic-Tac-Toe-project/blob/4674ffcc73200b617a9bc91abadd19a68b0b041f/spec/tic_tac_toe_spec.rb#L8-L12
Well after describe i did
```
 let(:board_test) { Board.new }
```   
and deleted the assignment you have done and It worked. Let me know if you have any different concept. (Always welcome.)
3. One test case is failing. Take a close look on the test, you will get the cause.
https://github.com/RoodzFernando/Tic-Tac-Toe-project/blob/4674ffcc73200b617a9bc91abadd19a68b0b041f/spec/tic_tac_toe_spec.rb#L24-L30
4. dependent variables should be created using let :point_up:  above method
https://github.com/RoodzFernando/Tic-Tac-Toe-project/blob/4674ffcc73200b617a9bc91abadd19a68b0b041f/spec/tic_tac_toe_spec.rb#L40-L51
___
##### Please fix the above :point_up: issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Slack](_Shakhawat Hossain_)

**Comment by RoodzFernando** [comment]
We have made all the changes required for this project.

File: spec/tic_tac_toe_spec.rb [hunk 1]
```diff
@@ -1,14 +1,75 @@
 # frozen_string_literal: true
 
-# require 'bin/main'
-require './lib/player'
+load File.expand_path('../bin/main', __dir__)
+require File.expand_path('../lib/player', __dir__)
+require File.expand_path('../lib/board', __dir__)
```
Comment (suretrust) [inline spec/tic_tac_toe_spec.rb:5]: - [ ]  Using `require_relative` for files in your directory instead of `require` is a ruby convention.

"Require should be used for external files, like gems, while require_relative should be used for referring to files within your directory."

File: spec/tic_tac_toe_spec.rb [hunk 2]
```diff
@@ -1,14 +1,75 @@
...
-  describe Player do
-    describe '#player_signs' do
-      it "Check if the player sign is 'X' or 'O':" do
-        allow($stdin).to receive(:gets).and_return('X' || 'O')
-      end
+  describe 'board_show' do
+    let(:board_test) { Board.new }
+    it 'shows the board' do
+      expect { board_show(board_test) }.to output(
+        "+---+---+---+\n"\
```
Comment (suretrust) [inline spec/tic_tac_toe_spec.rb:12]: It's best practice to be clear about what method you are describing. For instance, use the Ruby documentation convention of `.` (or `::`) when referring to a class method's name and `#` when referring to an instance method's name.
```
describe '.authenticate' do
describe '#admin?' do
```

File: spec/tic_tac_toe_spec.rb [hunk 3]
```diff
@@ -1,14 +1,75 @@
...
+                                                                    } .to_stdout
+    end
+  end
+
+  describe 'game_finish?' do
+    let(:board_test) { Board.new }
+    it 'Check if X wins the game' do
+      allow(board_test).to receive(:board).and_return(['X', 'X', 'X', 4, 5, 6, 'O', 'O', 9])
+      expect(game_finish?(board_test)).to eq('X')
+    end
```
Comment (suretrust) [inline spec/tic_tac_toe_spec.rb:58]: It's best practice to have the `it` block followed by a singular verb. `it 'checks if X wins'` is much better.

**Review by suretrust — APPROVED** [review]
Hi @RoodzFernando and @luis-novoa,

###  Amazing job! :raised_hands: No deal-breaker, let's get this merged. :+1: 

- [ ] It would be nice to note my comments on your code or even better, change them as appropriate. However, it's optional. :wink: 

_Good luck with your future projects and happy coding!_ :tada:
